server/server.ts

```diff
@@ -428,7 +429,263 @@ app.get('/api/compare/all', authMiddleware, async (req, res, next) => {
+    `;
+    const followingCountResp = await db.query(followingCountSql, [
+      comparatorId,
+    ]);
+    const [followingCount] = followingCountResp.rows;
+    const recommendationSql = `
+      select "filmTMDbId", "filmPosterPath", "rating", "liked"
+        from "filmLogs"
+        where "userId" = $1 and
+        "filmTMDbId" not in (select "filmTMDbId" from "filmLogs" where "userId" = $2) and
+        "rating" IS NOT NULL
+        order by "rating" desc
+        limit 4;
+    `;
+    const recommendationResp = await db.query(recommendationSql, [
+      comparatorId,
+      activeUserId,
+    ]);
+    const recommendations = recommendationResp.rows;
+    const recentReviewsSql = `
+      select "filmTMDbId", "filmPosterPath", "dateWatched", "review"
+        from "filmLogs"
+        where "userId" = $1 and
+        "review" IS NOT NULL
+        order by "dateWatched" desc
+        limit 3;
+    `;
+    const recentReviewsResp = await db.query(recentReviewsSql, [comparatorId]);
+    const recentReviews = recentReviewsResp.rows;
+    const overlappingWatchedSql = `
+      select COUNT(distinct "filmTMDbId") as "overlappingWatched"
+        FROM "filmLogs"
+        where "filmTMDbId" in (select "filmTMDbId" from "filmLogs" where "userId" = $1) and
+        "filmTMDbId" in (select "filmTMDbId" from "filmLogs" where "userId" = $2);
+    `;
+    const overlappingWatchedResp = await db.query(overlappingWatchedSql, [
+      activeUserId,
+      comparatorId,
+    ]);
+    const [overlappingWatched] = overlappingWatchedResp.rows;
+    const overlappingRatingsSql = `
+      SELECT COUNT(distinct t1."filmTMDbId") as "overlappingRatings"
+        FROM "filmLogs" AS t1
+        JOIN "filmLogs" AS t2 ON t1."filmTMDbId" = t2."filmTMDbId" AND t1."rating" = t2."rating"
+        WHERE t1."userId" = $1 AND t2."userId" = $2;
+    `;
+    const overlappingRatingsResp = await db.query(overlappingRatingsSql, [
+      activeUserId,
+      comparatorId,
+    ]);
+    const [overlappingRatings] = overlappingRatingsResp.rows;
+    const overlappingLikedSql = `
+      SELECT COUNT(distinct t1."filmTMDbId") as "overlappingLiked"
+        FROM "filmLogs" AS t1
+        JOIN "filmLogs" AS t2 ON t1."filmTMDbId" = t2."filmTMDbId" AND t1."liked" = t2."liked"
+        WHERE t1."userId" = $1 AND t2."userId" = $2;
+    `;
+    const overlappingLikedResp = await db.query(overlappingLikedSql, [
+      activeUserId,
+      comparatorId,
+    ]);
+    const [overlappingLiked] = overlappingLikedResp.rows;
+    res.json({
+      ...highestCorr,
+      ...username,
+      ...filmCount,
+      ...followingCount,
+      recommendations: [...recommendations],
+      recentReviews: [...recentReviews],
+      ...overlappingWatched,
+      ...overlappingRatings,
+      ...overlappingLiked,
+    });
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/compare/following', authMiddleware, async (req, res, next) => {
+  try {
+    const activeUserRatingsSql = `
```

Comment:
It's hard to tell for sure, but it looks like there is a fair amount of duplicated code between this endpoint and the previous one. You might want to extract it into separate functions.